spring-social-facebook/src/main/java/org/springframework/social/facebook/api/impl/AbstractFacebookOperations.java

```diff
@@ -73,14 +94,95 @@ public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttp
 		public boolean equals(Object o) {
 			if (this == o) return true;
 			if (o == null || getClass() != o.getClass()) return false;
-			OperationsRequestLoggingInterceptor that = (OperationsRequestLoggingInterceptor) o;
-			return id == that.id;
+			LoggingRequestsInterceptor that = (LoggingRequestsInterceptor) o;
+			return id.equals(that.id);
+		}
 
+		@Override
+		public int hashCode() {
+			return id.hashCode();
+		}
+	}
+
+	/**
+	 * https://developers.facebook.com/docs/graph-api/advanced/rate-limiting
+	 */
+	private class RateLimitRequestInterceptor implements ClientHttpRequestInterceptor {
+		private final String id;
+		private final int threshold = 80; //we shouldn't use more than this percentage
+
+		private static final String CALL_COUNT_PARAM = "call_count"; //
+		private static final String TOTAL_TIME_PARAM = "total_time";
+		private static final String TOTAL_CPU_TIME_PARAM = "total_cputime";
+		private static final String X_APP_USAGE_PARAM = "X-App-Usage";
+
+		public RateLimitRequestInterceptor(int id) {
+			this.id = "rateLimit-interceptor-" + id;
+		}
+
+		public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttpRequestExecution execution) throws IOException {
+			ClientHttpResponse response = execution.execute(request, body);
+			logger.debug("Request: " + request.getMethod().name() + " " + request.getURI().toString());
+			logger.debug("Response: " + response.getRawStatusCode() + " " + response.getStatusText());
+			if (mightGetRateLimited(response.getHeaders())) {
+				logger.warn("Approaching API usage.");
+				throw new FacebookThresholdLimitReachedException();
+			}
+			return response;
+		}
+
+		/**
+		 * The values for callCount, totalTime and totalCpuTime are whole numbers representing the percentage used
+		 * values for each of the metrics. When any of these metrics exceed 100 the app will be rate limited.
+		 * So this method verifies if the threshold value was reached
+		 *
+		 * @param headers
+		 * @return true when the threshold was reached, false otherwise.
+		 */
+		private boolean mightGetRateLimited(HttpHeaders headers) {
+			logger.debug("Response Headers: " + headers.toSingleValueMap());
+			if (!headers.containsKey(X_APP_USAGE_PARAM)) {
+				return false;
+			}
+
+			List<String> appUsageParams = headers.get(X_APP_USAGE_PARAM);
+			if (appUsageParams == null || appUsageParams.isEmpty()) {
+				return false;
+			} else {
+				logger.debug("APP Usage params: " + appUsageParams);
+			}
+
+			if (readFrom(headers, CALL_COUNT_PARAM) >= threshold
+					|| readFrom(headers, TOTAL_TIME_PARAM) >= threshold
```

Comment:
I don't understand the `total_time >= 80` or `total_cputime >= 80` part, these params are supposed to represent how many seconds have passed since the request has been made right?